### PR TITLE
Processing data in models

### PR DESCRIPTION
**Pull request by cynthiainga**
**Project requirements:**

- [ ] Set up associations between models
- [ ] Create custom methods
    - Users
      - A method that returns the 3 most recent posts for a given user.
    - Posts
      - A method that updates the posts counter for a user.
      - A method that returns the 5 most recent comments for a given post.
    - Comments
      - A method that updates the comments counter for a post.
    - Likes
      - A method that updates the likes counter for a post.
- [ ] Created a current_user method that returns the first user from a database.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -1,3 +1,16 @@
...
+
+  private
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+
+  def recents_comments
+    comments.order(created_at: :desc).limit(5)
+  end
```
Comment (mikethreels) [inline app/models/post.rb]: - [ ] in this setup the `recents_comments` is a private method as well because it is below the `private` keyword

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Candy, 

Good job your code is spot on!
There is one small error that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (cynthiainga) [reply app/models/post.rb]: Ohh let me change it now, thanks for the review @mikethreels

Comment (cynthiainga) [reply app/models/post.rb]: @mikethreels So am I supposed to put it before the `private` keyword??

Comment (mikethreels) [reply app/models/post.rb]: Hi @cynthiainga,
There are 2 ways you can go about fixing it, you can do what you already suggested or use the `public` keyword above the method you want to make public

Comment (cynthiainga) [reply app/models/post.rb]: Okay, I see, thank you @mikethreels, let me fix it now and ask for the second review

**Review by mikethreels — APPROVED** [review]
Hi Candy, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by cynthiainga** [comment]
Thanks @mikethreels  for the review, I really appreciate it :pray:

**Pull request merged**